### PR TITLE
First pass metabot visualization generation capability

### PR DESCRIPTION
**Pull request by markbastian**
This is a first generation pass of adding automatic "interesting" visualizations to metabot queries.

This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/29848)

### CODE REVIEW

**Review by alxnddr — APPROVED** [review]
Approve FE

File: src/metabase/metabot/util.clj [hunk 1]
```diff
@@ -205,16 +205,24 @@
...
       {:prompt_type prompt_task}))))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;; Results Processing ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+(defn select-all?
+  "Is this a simple SELECT * query?"
+  [sql]
+  (str/starts-with?
+   (str/replace sql #"\s+" "")
+   "SELECT*"))
```
Comment (metamben) [inline src/metabase/metabot/util.clj:224]:
```suggestion
  (some? (re-find #"(?i)^select\s*\*" sql)))
```

File: src/metabase/metabot/util.clj [hunk 2]
```diff
@@ -245,3 +253,21 @@
...
+  "Given a response from the LLM, map this to visualization settings. Default to a table."
+  [{:keys [display description visualization_settings]}]
+  (let [display (keyword display)
+        {:keys [x-axis y-axis]} visualization_settings]
+    (case display
+      (:line :bar :area :waterfall) {:display                display
+                                     :name                   description
+                                     :visualization_settings {:graph.dimensions [x-axis]
+                                                              :graph.metrics    y-axis}}
+      (:scalar) {:display                display
```
Comment (metamben) [inline src/metabase/metabot/util.clj:267]:
```suggestion
      :scalar {:display                display
```

**Review by metamben — APPROVED** [review]
BE LGTM

File: src/metabase/api/metabot.clj [hunk 3]
```diff
@@ -24,6 +24,29 @@
...
 
+(defn- infer-sql-or-throw
+  "An http-friendly version of infer-sql that throws a useful error if it fails to produce sql."
+  [context question]
+  (or
+   (metabot/infer-sql context)
+   (throw
+    (let [message (format
+                   "Query '%s' didn't produce any SQL. Perhaps try a more detailed query."
+                   question)]
```
[the remark this answers is not in the document]
Comment (markbastian) [reply src/metabase/api/metabot.clj:35]: Let me think about it, but I think I'll leave as-is for now. I am assuming at some point we will want to. Good callout.

**Pull request merged**